### PR TITLE
added spoolman support

### PR DESCRIPTION
**Pull request by Surion79**
I added spoolman integration to klippain.

Source: https://github.com/Donkie/Spoolman

I tried using the configuration with the provided default settings (in my case i went with docker installation on the pi) It works with localhost to point to itself in browsing the spoolman webinterface and the integration in mainsail.

### CODE REVIEW

**Comment by Benoitone** [comment]
I also added it in the Happy Hare branch :)

**Comment by Surion79** [comment]
it would be easier, if you check this, instead of me finding in the dozens of modified files of your PR where you integrated spoolman :)
(and it wouild be helpful to have one function per PR and don't mix them together)

**Comment by Surion79** [comment]
i think you are missing the moonraker.conf include for spoolman to work.
And i don't know if it is a good idea to create files which include only a single include to a different file.

**Comment by roykrikke** [comment]
+1

File: moonraker/spoolman.conf [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# this is a default configuration pointing the default installation on the printer itself
+# you can set an own config by excluding this file, copy its content in moonraker.conf and change the values
+[spoolman]
+server: http://localhost:7912
```
Comment (Benoitone) [inline moonraker/spoolman.conf:4]: for me I must use ip instead of localhost to work properly :(

File: user_templates/moonraker.conf [hunk 2]
```diff
@@ -31,6 +31,9 @@
 # [include moonraker/z_calibration.conf]
 # -----------------------------------------------------------------
 
+##### Spoolman configuration file  -----------------
+# [include moonraker/spoolman.conf]
+# -----------------------------------------------------------------
```
Comment (Benoitone) [inline user_templates/moonraker.conf:36]:
```suggestion
##### Spoolman configuration file  -----------------
# [include moonraker/spoolman.conf]
#[spoolman]
#server: http://SpoolmanServerIP:7912
# -----------------------------------------------------------------
```

**Comment by Frix-x** [comment]
I'll close this PR since everything is now properly managed in the MMU management PR #371

**Pull request closed without merging**